graphics/bezier_curve.py

```diff
@@ -0,0 +1,107 @@
+# https://en.wikipedia.org/wiki/B%C3%A9zier_curve
+# https://www.tutorialspoint.com/computer_graphics/computer_graphics_curves.htm
+
+from typing import List, Tuple
+import scipy.special as sp
+
+
+class BezierCurve:
+    """
+    A class to generate bezier curves for a given set of points. The implementation works only for 2d points.
+    """
+
+    def __init__(self, list_of_points: List[Tuple]):
+        """
+        The constructor of the Bezier Curve class.
+            list_of_points: List of tuples on which to interpolate.
+        """
+        self.list_of_points = list_of_points
+        self.degree = (
+            len(list_of_points) - 1
+        )  # degree of the curve produced will be n - 1
+
+    def basis_function(self, t: float) -> List[float]:
+        """
+        The function to cater the need for the basis function.
+            t: the 1-d co-ordinate at which to evaluate the basis function(s).
+            returns a returns list of floating point numbers, a number at index i represents the value of
+            basis function i at t.
+
+        >>> curve1 = BezierCurve([(1,1), (1,2)])
+        >>> curve1.basis_function(0)
+        [1.0, 0.0]
+        >>> curve1.basis_function(1)
+        [0.0, 1.0]
+        """
+        output_values: List[float] = list()
```

Comment:
Should we `assert 0 <= t <= 1` ?

We do not need type annotations outside of the function signatures.  Let's do this a list comprehension...
```
x_y = tuple([comb(self.degree, i) * ((1 - t) ** (self.degree - i)) * (t ** i)
             for i in range(len(self.list_of_points))])
```

Reply:
Agreed with the assert part. Don't you think doing this as a list comprehension reduces the readability as there is a lot going on inside?